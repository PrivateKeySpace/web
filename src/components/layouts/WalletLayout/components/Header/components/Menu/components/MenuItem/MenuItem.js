import React from 'react'
import { compose, setDisplayName, pure } from 'recompose'
import { css } from 'aphrodite/no-important'
import classNames from 'classnames'
import tx from '../../../../../../../../../theme/styles'
import sx from './styles'

const displayName = 'components/layouts/WalletLayout/components/Header/components/Menu/components/MenuItem'

const enhance = compose(
  setDisplayName(displayName),
  pure
)

function MenuItem (props) {
  const { icon, children, onClick } = props
  return (
    <div
      className={css(tx.flex, tx.alignItemsCenter, tx.px3, tx.ml1, tx.width100, tx.cursorPointer, sx.menuItem)}
      onClick={onClick}
    >
      {
        icon &&
        <span className={classNames(`lnr lnr-${icon}`, css(tx.fontSizes1, tx.mr3))} />
      }
      <div className={css(tx.textTransformUppercase, sx.menuItemText)}>{children}</div>
    </div>
  )
}

export default enhance(MenuItem)
