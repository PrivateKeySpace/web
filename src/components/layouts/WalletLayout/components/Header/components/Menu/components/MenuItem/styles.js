import { StyleSheet } from 'aphrodite/no-important'
import { colors } from '../../../../../../../../../theme/constants'

const styles = StyleSheet.create({
  menuItem: {
    height: 34,
    color: '#999999',
    ':hover': {
      color: colors.white
    }
  },

  menuItemText: {
    fontSize: 14
  }
})

export default styles
