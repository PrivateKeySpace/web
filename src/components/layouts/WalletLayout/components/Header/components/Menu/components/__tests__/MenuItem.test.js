import React from 'react'
import { StyleSheetTestUtils } from 'aphrodite'
import { render } from 'enzyme'
import MenuItem from '../MenuItem'

beforeEach(() => {
  StyleSheetTestUtils.suppressStyleInjection()
})

afterEach(() => {
  StyleSheetTestUtils.clearBufferAndResumeStyleInjection()
})

it('renders without crashing and matches snapshot', () => {
  const footer = render(
    <MenuItem />
  )
  expect(footer).toMatchSnapshot()
})
