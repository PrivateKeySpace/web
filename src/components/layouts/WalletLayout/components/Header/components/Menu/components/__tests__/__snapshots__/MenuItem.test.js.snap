// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing and matches snapshot 1`] = `
<div
  class="flex_jro6t0-o_O-alignItemsCenter_1moixrt-o_O-px3_1rzpu44-o_O-ml1_1dn7v29-o_O-width100_zdxht7-o_O-cursorPointer_13xlah4-o_O-menuItem_k33g8x"
>
  <div
    class="textTransformUppercase_4efw5a-o_O-menuItemText_f9e8qy"
  >
    
  </div>
</div>
`;
